### PR TITLE
Splunk sid

### PR DESCRIPTION
**Pull request by maimorag**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:[ link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-4281)

## Description
Output the SID in splunk-search, regardless of whether the command runs/ends successfully.

## Minimum version of Cortex XSOAR
- [x] 6.0.0

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 1]
```diff
@@ -1952,15 +1952,22 @@ def build_search_query(args):
...
+    number_of_results = len(parsed_search_results)
 
     if args.get('update_context', "true") == "true":
         ec['Splunk.Result'] = parsed_search_results
         if len(dbot_scores) > 0:
             ec['DBotScore'] = dbot_scores
         if status_res:
-            ec['Splunk.JobStatus(val.SID && val.SID === obj.SID)'] = status_res.outputs
+            ec['Splunk.JobStatus(val.SID && val.SID === obj.SID)'] = {
+                **status_res.outputs, **{'total_results': number_of_results}}
```
Comment (DeanArbel) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:1965]: Why do you need to unpack `total_results`?
You can provide it without unpacking it as one of the keys.

File: Packs/SplunkPy/ReleaseNotes/3_0_5.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### SplunkPy
+- Added the sid for any **splunk-search** command status for quicker results return.
```
Comment (DeanArbel) [inline Packs/SplunkPy/ReleaseNotes/3_0_5.md:4]:
```suggestion
- Added the *sid* output to the **splunk-search** command.
```

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 3]
```diff
@@ -1952,15 +1952,22 @@ def build_search_query(args):
...
             ec['DBotScore'] = dbot_scores
         if status_res:
-            ec['Splunk.JobStatus(val.SID && val.SID === obj.SID)'] = status_res.outputs
+            ec['Splunk.JobStatus(val.SID && val.SID === obj.SID)'] = {
+                **status_res.outputs, 'total_results': number_of_results}
+    if job_id and not status_res:
+        status = 'DONE' if (number_of_results > 0) else 'NO RESULTS'
+        ec['Splunk.JobStatus'] = [{'sid': job_id,
+                                   'total_results': number_of_results,
+                                   'Status': status}]
```
Comment (DeanArbel) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:1970]: Please add the new fields to the yml file. Please change `sid` to `SID` as it currently exists in the yml output.
The name `total_results`  doesn't match the existing standard either. `TotalResults` will be better.
image

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 4]
```diff
@@ -1952,15 +1952,22 @@ def build_search_query(args):
...
         ec['Splunk.Result'] = parsed_search_results
         if len(dbot_scores) > 0:
             ec['DBotScore'] = dbot_scores
         if status_res:
-            ec['Splunk.JobStatus(val.SID && val.SID === obj.SID)'] = status_res.outputs
+            ec['Splunk.JobStatus(val.SID && val.SID === obj.SID)'] = {
+                **status_res.outputs, 'TotalResults': number_of_results}
+    if job_id and not status_res:
+        status = 'DONE' if (number_of_results > 0) else 'NO RESULTS'
+        ec['Splunk.JobStatus'] = [{'SID': job_id,
```
Comment (DeanArbel) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:1968]: The `ec` should containt the `dt` as well, so it'll merge properly with any previous results, i.e.
```suggestion
        ec['Splunk.JobStatus(val.SID && val.SID === obj.SID)'] = [{'SID': job_id,
```

**Review by DeanArbel — APPROVED** [review]
Looks good, please update the release notes to match the addition of the values.

**Comment by maimorag** [comment]
updating description for yml file, therefore there is validation error

**Comment by maimorag** [comment]
Updating the description of "splunk-search" in the yml file, therefore there is a validation error of BC (BC102).

**Review by DeanArbel — APPROVED** [review]

**Pull request merged**